crates/mem-pool/src/pool.rs

```diff
@@ -454,6 +461,20 @@ impl MemPool {
         Self::package_mem_block(&self.mem_block, output_param)
     }
 
+    fn get_polyjuice_creator_id(&mut self) -> Result<Option<u32>> {
+        let snap = self.mem_pool_state.load();
+        let state = snap.state()?;
```

Comment:
Move these to line to the if branch, we only need the `state` when `if self.polyjuice_creator_id.is_none()` returns true.